ejerciciosPilas/src/actores/libroPrimaria/Charco.ts

```diff
@@ -1,7 +1,26 @@
-/// <reference path="../ActorAnimado.ts"/>
-class Charco extends ActorAnimado {
+/// <reference path = "../../../../bower_components/pilasweb/dist/pilasweb.d.ts"/>
+/// <reference path="../ActorColisionable.ts"/>
+/// <reference path="../../actores/libroPrimaria/Coty.ts"/>
+/// <reference path="../../comportamientos/SaltarAnimado.ts"/>
+/// <reference path="../../comportamientos/Hundir.ts"/>
+
+class Charco extends ActorColisionable {
+
     constructor() {
         super(0, 0, { grilla: 'actor.charco.png' });
         this.definirAnimacion("parado", [0], 6, true);
     }
+
+    afecta() {
+        return [Coty];
+    };
+
+    ignora() {
+        return [SaltarAnimado];
```

Comment:
Esto no se si me convence. Creo que prefiero delegar la condición de colisión al comportamiento del actor, y que Saltar devuelva false y el Dibujar (y otros) true.

Reply:
Bueno, creo que llegaste a lo que me hace ruido de la propuesta de @51lv3str1 .
Fijate que también sucede en el método `afecta()`, ahí también debería hacerse.

Pero fijense cómo queda yendo a full con la propuesta del Rasta:

```Typescript
class Charco extends ActorColisionable {
     override esAfectado(actor){
          return actor.seHundeEnCharco()
     }
     override debeIgnorarComportamiento(comportamiento){
        return !comportamiento.debeProducirColisionConCharco()
     }
}
```

Parece como que queda re cheto, pero ese doble dispatch me obliga a definir en cada actor y comportamiento de una escena donde hay un ActorColisionable, métodos para decidir en cada uno de los otros actores si cada uno de ellos debe colisionar con el que te jedi en cuestion. Y además también lo mismo con cada comportamiento. O sea, ponele que haya obstáculos, voy a tener que definir los métodos `Actor>>teChocasConObstaculo` y `Comportamiento>>debeProducirColisionConObstáculo` (4 métodos en total como mínimo: los que devuelven true y los que devuelven false)

Usar polimorfismo con double dispatch es alto overkill para este feature del ActorColisionable. Pucha, estoy diciendo que prefiero un instance of por sobre polimorfismo en una plataforma pública. 

Otra cosa que me tira acá también es no sobrediseñar.

No sé, ¿qué opinan?

Reply:
Concuerdo con vos! Yo no quiero que "charco" aparezca en el nombre de un msj, sino como parámetro.
Lo que me causaba ruido, es que `SaltarAnimado` es algo propio de Coty. Y todo bien con que el charco quede acoplado a Coty, por medio de personajes a los que afecta (aunque en un ideal creo que tampoco haría falta), pero ya saber qué comportamiento DE COTY es con el que no tengo que hacer nada, me parece que acopla mucho y no sé si quiero romper el encapsulamiento de `actor.comportamiento_actual`.
Esa validación es lo que dejaría como responsabilidad del actor, onda:
```
teEstoyPorColisionar(actor: Actor) {
  ... validaciones y cosas ...
  actor.hacerPorColision(comportamiento)
}
```

Y que en ese `hacerPorColision` Coty lo delegue al comportamiento actual o algo así para que si está saltando no se ejecuta y sino sí.

Qué opinan?